python/lsst/ctrl/bps/bps_config.py

```diff
@@ -86,6 +86,11 @@ def __init__(self, other, search_order=None):
             self.search_order = search_order
             self.formatter = BpsFormatter()
 
+        # Make sure search sections exist
+        for key in self.search_order:
+            if not Config.__contains__(self, key):
+                self[key] = {}
+
```

Comment:
you could use `Config.merge` for this;
`self.merge({k, {} for key in self.search_order}`

Reply:
merge doesn't like regular dict.  Filed a ticket DM-27473.  When that ticket fixed, will make change to bps_config either in this ticket or later ticket.